### PR TITLE
Fix 2 errors in phenopacket import

### LINKED ISSUE #3734: Import phenopacket can crash - on empty json responses?
```
2022-12-20 15:20:00,503 - scout.server.app - ERROR: Exception on /cust002/F0007523/phenotype_import [POST] [in /venv/lib/python3.8/site-packages/flask/app.py:1741]
Traceback (most recent call last):
 File "/venv/lib/python3.8/site-packages/flask/app.py", line 2525, in wsgi_app
   response = self.full_dispatch_request()
 File "/venv/lib/python3.8/site-packages/flask/app.py", line 1822, in full_dispatch_request
   rv = self.handle_user_exception(e)
 File "/venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
   return cors_after_request(app.make_response(f(*args, **kwargs)))
 File "/venv/lib/python3.8/site-packages/flask/app.py", line 1820, in full_dispatch_request
   rv = self.dispatch_request()
 File "/venv/lib/python3.8/site-packages/flask/app.py", line 1796, in dispatch_request
   return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
 File "/home/worker/app/scout/server/blueprints/cases/views.py", line 409, in phenotype_import
   phenopacket = phenopacketapi.get_hash(phenopacket_hash)
 File "/home/worker/app/scout/server/extensions/phenopacket_extension.py", line 109, in get_hash
   json_content = json_response.get("content")[0]
TypeError: 'NoneType' object is not subscriptable
```
after doing:
Screenshot 2022-12-20 at 15 21 02
At less prio, the button there should not be labeled "Upload", not to confuse it with the first (file upload) alternative. Perhaps "Retrieve", "Fetch" or the like.

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3734 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/extensions/phenopacket_extension.py [hunk 1]
```diff
@@ -101,10 +101,10 @@ def get_hash(self, phenopacket_hash):
 
         query = f"{self.url}/api/v1/phenopacket?hash={phenopacket_hash}"
 
-        try:
-            json_response = get_request_json(query)
-        except Exception as e:
-            LOG.warning(e)
+        json_response = get_request_json(query)
```
Comment (northwestwitch) [inline scout/server/extensions/phenopacket_extension.py:104]: Try/catch block is already present in [get_request_json function](https://github.com/Clinical-Genomics/scout/blob/a22110f94e40fd11f1f343559c8ff6e2afbe0979/scout/utils/scout_requests.py#L57)

File: scout/server/blueprints/cases/views.py [hunk 2]
```diff
@@ -397,8 +397,14 @@ def phenotype_import(institute_id, case_name):
     institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
     user_obj = store.user(current_user.email)
     case_url = url_for(".case", institute_id=institute_id, case_name=case_name)
+    phenopacket = None
 
     phenopacket_file = request.files["phenopacket_file"]
+    phenopacket_hash = request.form.get("phenopacket_hash")
+
+    if not phenopacket_file or phenopacket_hash:
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/views.py:405]: This is fixing problem n.2

File: scout/server/extensions/phenopacket_extension.py (hunk 1 above)
Comment (dnil) [reply scout/server/extensions/phenopacket_extension.py:104]: Right, good catch - in particular that function did not re-raise the exception like the main one did before.. 🙄

File: scout/server/blueprints/cases/views.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/views.py:405]: 🥇

File: scout/server/blueprints/cases/templates/cases/phenotype.html [hunk 3]
```diff
@@ -327,7 +327,7 @@ <h5 class="modal-title" id="importPhenotypeLabel">Import phenotype case</h5>
               <p class="help-block">How do I format my <a href="https://phenopacket-schema.readthedocs.io/en/latest/index.html" rel="noopener" target="_blank">phenopacket</a> JSON file?</p>
             </div>
             <div class="col-4">
-             <button type="submit" class="btn btn-secondary">Upload</button>
+             <button type="submit" class="btn btn-secondary">Import</button>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/phenotype.html:330]: That will do!

File: CHANGELOG.md [hunk 4]
```diff
@@ -23,8 +23,9 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Update ClinVar submission ID form
 - Handle connection timeout when sending requests requests to external web services
 - Validate any ClinVar submission regardless of its status
+- Errors in Phenomizer import feature
```
Comment (dnil) [inline CHANGELOG.md:26]:
```suggestion
- Empty Phenopackets import crashes
```

File: scout/server/extensions/phenopacket_extension.py [hunk 5]
```diff
@@ -101,10 +101,11 @@ def get_hash(self, phenopacket_hash):
...
+
+        if json_response.get("message"):
+            LOG.warning(json_response["message"])
```
Comment (dnil) [inline scout/server/extensions/phenopacket_extension.py:107]: 👍

**Review by dnil — COMMENTED** [review]
Thank you! It does not seem to work for connecting to the backend now though? 😊

**Comment by dnil** [comment]
I tried restarting the backend, but all seems good there: I can e.g. get a packet like https://phenopacket-stage.scilifelab.se/api/v1/phenopacket?hash=3449c9e5e332f1dbb81505cd739fbf3f

However the same will not load: 
Screenshot 2022-12-21 at 08 31 39

File: scout/server/blueprints/cases/views.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/views.py:405]: Ahem, its here..

Comment (dnil) [reply scout/server/blueprints/cases/views.py]:
```suggestion
    if not phenopacket_file and not phenopacket_hash:
```

**Review by dnil — APPROVED** [review]

**Pull request merged**